Bucketeer/Sources/BucketeerSDK.swift

```diff
@@ -316,8 +317,16 @@ private extension BucketeerSDK {
         if let eventsCount = eventsCount, eventsCount < config.minEventsPerRequest {
             return
         }
-        eventStore.fetch(limit: config.maxEventsPerRequest) { [eventRegisterer] eventEntities in
-            eventRegisterer.registerEvents(eventEntities: eventEntities)
+        self._registerEvents()
+    }
+
+    func _registerEvents(completion: (() -> Void)? = nil) {
+        self.registerEventsQueue.sync {
+            eventStore.fetch(limit: config.maxEventsPerRequest) { eventEntities in
+                self.eventRegisterer.registerEvents(eventEntities: eventEntities) { _ in
+                    completion?()
+                }
+            }
```

Comment:
Hi, I don't think it will be thread-safe even if I make this a serial queue sync, because the process inside is changing to another thread.
In this case, I guess you want to lock the following
lock start
- Get eventEntities from DB
- Execute registerEventsAPI
- Up to deleting eventEntities
lock end

It seems better to use NSLock, DispatchSemaphore or objc_sync_enter, what do you think?

ex)
```
let lock = NSLock()
...
lock.lock()
registerEventsPoller.stop()
eventStore.fetch(limit: config.maxEventsPerRequest) { eventEntities in
    self.eventRegisterer.registerEvents(eventEntities: eventEntities) { _ in
        self.startRegisterEventsPolling()
        lock.unlock()
    }
}
```